### PR TITLE
DXCDT-328: Update documentation theme

### PR DESCRIPTION
**Pull request by ewanharris**
### 🔧 Changes

This replaces the theme used by the Jekyll docs with the [just-the-docs](https://just-the-docs.github.io/just-the-docs/) theme that's built for documentation usage.

A summary of the commits:

- 6c40d9a69dcead4a8e06c76a78776992a8b471f0, 4f3f84100a2c71f494992689631c2273eda467e4, 23b85eb56c22d885a72a9852ff5663722cdab64d - Getting the theme working, removed the `_includes` as they were causing issues, changed to use via `remote_theme` as [GitHub pages requires that](https://docs.github.com/en/pages/setting-up-a-github-pages-site-with-)
- 9b9b505a7c5a5f77d4adbdfefc9e9df603611745, 4dce5adf551d82a06d392d64f4abbbf52286e3d5, 516e6fbf5e11903b706feb3e3b86f9b5b058f94d, 89c80b47a57739156d18bf6295d96ac65c36eca6,d1a4a8b088fbedb4e2a16c02915eef2ef0fcf852 - Changes to the doc-gen script that improve the structure of the navigation sidebar to allow us to nest runnable commands under their non-runnable parent
- 002cc50d0b848a579280e9de03eb6c7aa7cd122f - Adds a "Auth0 CLI on GitHub" link to the header
- 5db45a02919ffdb6d23a307ad1d81719744d3554 - Generates the docs
- eec72e36b60465acef0605b6cd75df275d97c016 - Removes the file that generates the search content as we're no longer using a local theme

The sidebar navigation is a best effort of what we could get to right now, but is almost certainly not perfect. just-the-docs [has a limit of 3 levels](https://just-the-docs.github.io/just-the-docs/docs/navigation-structure/#children-with-children) for the nesting of the navigation currently, which doesn't align well with the structure of the commands we have. Given this limitation (and in the interest of time) we chose to make non-runnable subcommands a top level in the navigation with the runnable subcommands underneath. So for example (some commands missing for brevity) 

- `auth0 logs`
	-	`auth0 logs list`
- `auth0 logs streams`
	- `auth0 logs streams delete`
	- `auth0 logs streams list`
- `auth0 logs streams create`
	- `auth0 logs streams create datadog`

A potential improvement might be to try and nest the `auth0 logs streams create` command underneath `auth0 logs streams`, but with the current version of just-the-docs we are not able to nest `auth0 logs streams` (and its children) underneath `auth0 logs` due to the aforementioned limitation.

### 🔬 Testing

The site is live at https://ewanharris.github.io/auth0-cli/  (will be removed once we've merged and published)

### 📝 Checklist

- [x] All new/changed/fixed functionality is covered by tests (or N/A)
- [x] I have added documentation for all new/changed functionality (or N/A)

### CODE REVIEW

File: docs/_config.yml [hunk 1]
```diff
@@ -11,7 +11,11 @@ github_username: auth0
...
+remote_theme: just-the-docs/just-the-docs@v0.4.0.rc3
 
-header_pages:
-  -
+search_enabled: true
+favicon_ico: "/assets/images/favicon.png"
+
+aux_links:
+  "Auth0 CLI on GitHub":
+    - "//github.com/auth0/auth0-cli"
```
Comment (sergiught) [inline docs/_config.yml:21]:
```suggestion
    - "//github.com/auth0/auth0-cli"

```

Just adding a new line at the end.

File: docs/_includes/favicons.html [hunk 2]
```diff
@@ -1 +0,0 @@
-<link rel="icon" type="image/png" href="/favicon.png">
```
Comment (sergiught) [inline docs/_includes/favicons.html:1]: The current docs do have a favicon included: https://github.com/auth0/auth0-cli/blob/main/docs/_includes/favicon.png. I'm not sure about removing this file tbh 🤔 Or is this coming from the theme itself?

**Review by sergiught — COMMENTED** [review]
Solid contribution! 💯

File: docs/_includes/favicons.html (hunk 2 above)
Comment (ewanharris) [reply docs/_includes/favicons.html:1]: This should be configured in the [config](https://github.com/auth0/auth0-cli/pull/615/files#diff-f64371f310bcd3bbf6ead5827d929a4eb13b7516806d6658485a996e39eb2556R17) now but it seems like the live site doesn't have it (although it did work locally at one point). I'll take a look

Comment (ewanharris) [reply docs/_includes/favicons.html:1]: Ah shoot, it used to work before I reverted to rc3 (for a fix for the padding on the search box) as it was added in rc4 - https://github.com/just-the-docs/just-the-docs/blob/main/CHANGELOG.md#new-features.

I'll see if I can add a custom head file that will tide us over til there's a 0.3.0 release or an rc5

Comment (ewanharris) [reply docs/_includes/favicons.html:1]: Resolved in 96a4182315470a94127c2e67d646e50f8e2a9882

**Review by Widcket — APPROVED** [review]

**Pull request closed without merging**